sources/ElkArte/AttachmentsDirectory.php

```diff
@@ -0,0 +1,876 @@
+		elseif (substr($modSettings['attachmentUploadDir'], 0, strlen(BOARDDIR)) != BOARDDIR)
+		{
+			unset($context['file_tree'][strtr(BOARDDIR, array('\\' => '/'))]['contents']['attachments']);
+			$context['file_tree'][strtr($modSettings['attachmentUploadDir'], array('\\' => '/'))] = array(
+				'type' => 'dir',
+				'writable_on' => 'restrictive',
+			);
+		}
+	}
+
+	/**
+	 * Check and create a directory automatically.
+	 *
+	 * @param bool $is_admin_interface
+	 */
+	public function automanage_attachments_check_directory($is_admin_interface = false)
+	{
+		global $modSettings, $context;
+
+		// Not pretty, but since we don't want folders created for every post.
+		// It'll do unless a better solution can be found.
+		if ($this->checkNewDir($is_admin_interface) === false)
+		{
+			return;
+		}
+
+		// Get our date and random numbers for the directory choices
+		$year = date('Y');
+		$month = date('m');
+
+		$rand = md5(mt_rand());
+		$rand1 = $rand[1];
+		$rand = $rand[0];
+
+		if (!empty($this->baseDirectories) && !empty($this->useSubdirectories))
+		{
+			if (!is_array($this->baseDirectories))
+			{
+				$this->baseDirectories = Util::unserialize($this->baseDirectories);
+			}
+
+			$base_dir = array_search($modSettings['basedirectory_for_attachments'], $this->baseDirectories);
+		}
+		else
+		{
+			$base_dir = 0;
+		}
+
+		$basedirectory = !empty($this->useSubdirectories) ? $modSettings['basedirectory_for_attachments'] : BOARDDIR;
+
+		// Just to be sure: I don't want directory separators at the end
+		$sep = (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') ? '\/' : DIRECTORY_SEPARATOR;
+		$basedirectory = rtrim($basedirectory, $sep);
+
+		switch ($this->automanage_attachments)
+		{
+			case self::AUTO_SEQUENCE:
+				$this->initLastDir($base_dir);
+				$updir = $basedirectory . DIRECTORY_SEPARATOR . 'attachments_' . $this->last_dirs[$base_dir];
+				break;
+			case self::AUTO_YEAR:
+				$updir = $basedirectory . DIRECTORY_SEPARATOR . $year;
+				break;
+			case self::AUTO_YEAR_MONTH:
+				$updir = $basedirectory . DIRECTORY_SEPARATOR . $year . DIRECTORY_SEPARATOR . $month;
+				break;
+			case self::AUTO_RAND:
+				$updir = $basedirectory . DIRECTORY_SEPARATOR . (empty($this->useSubdirectories) ? 'attachments-' : 'random_') . $rand;
+				break;
+			case self::AUTO_RAND2:
+				$updir = $basedirectory . DIRECTORY_SEPARATOR . (empty($this->useSubdirectories) ? 'attachments-' : 'random_') . $rand . DIRECTORY_SEPARATOR . $rand1;
+				break;
+			default:
+				$updir = '';
+		}
+
+		if (!is_array($modSettings['attachmentUploadDir']))
+		{
+			$modSettings['attachmentUploadDir'] = Util::unserialize($modSettings['attachmentUploadDir']);
+		}
+
+		if (!in_array($updir, $modSettings['attachmentUploadDir']) && !empty($updir))
+		{
+			$outputCreation = automanage_attachments_create_directory($updir);
+		}
+		elseif (in_array($updir, $modSettings['attachmentUploadDir']))
+		{
+			$outputCreation = true;
+		}
+		else
+		{
+			$outputCreation = false;
+		}
+
```

Comment:
Since I'm being a total PITA, just set outputCreation to false before the first if and drop the last else, that else is just there to ensure var initialization, so initialize it to begin with.